### PR TITLE
Simplify null parameter checks for all constructors

### LINKED ISSUE #153: 🚧Simplify null parameter checks for all constructors
### I have done the items below . . .

- [X] I have updated the title by replacing the '**_<title_**>' section.

### Description

Using a guard pattern using a static class to wrap and simplify the throwing of exceptions, create the guard class and implement throughout the code base.

Great video [here](https://www.youtube.com/watch?v=MHIheQ2_Yb4&t=318s) to demonstrate this.

### Acceptance Criteria

**This issue is finished when:**
- [x] Guard class created with unit tests
- [x] Use `[CallerArgumentExpression]` for method params to help improve the exception message
- [x] Replace `EnsureThat.StringParamIsNotNullOrEmpty()` exception message with _**The string parameter must not be null or empty.**_
    - This will require changes to unit tests.
- [x] Unit tests added
- [x] All unit tests pass

### ToDo Items

- [x] Draft pull request created and linked to this issue
- [X] Priority label added to issue (**_low priority_**, **_medium priority_**, or **_high priority_**)
- [X] Issue linked to the proper project
- [X] Issue linked to proper milestone
- [x] Unit tests have been written and/or adjusted for code additions or changes
- [x] All unit tests pass

### Issue Dependencies

_No response_

### Related Work

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### LINKED ISSUE #176: 🧪QA Testing Velaptor
### To Do List:

- [X] I have linked all issues that are getting tested for this release
- [X] QA testing label added to issue (🧪qa testing)

### Version Of Velaptor Being Tested

v1.0.0-preview.7

### Relevant Information

This release requires testing of 1 new feature and also added testing functionality for existing sound behavior.

### **Added features:**
1. Rendering of rectangles
    - This new feature adds the ability for a user to easily render rectangles to the screen with various attributes.
    These attributes are:
        - Full and empty rectangles
        - Border thickness of empty rectangles
        - Width and height
        - Position
        - Corner radius for each corner independently
        - If the type of gradient is set to none, the color of the full rectangle and the border color of the empty rectangle
        - Horizontal color gradients for full and empty rectangles
        - Vertical color gradients for full and empty rectangles
        - Starting color of either gradient type
        - Ending color of either gradient type

### Issues To Test

- [x] #153
    - Just make sure that all other testing application functionality works as normal 
- [x] #75 
    - This will involve using the new **rectangle testing scene** that has been added to the testing application
- [x] #131 
    - This will involve using the new **sound testing scene** that has been added to the testing application

### QA Testing Dependencies

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by CalvinWilkinson**
<h1 style="font-weight:bold" align="center">Preview Feature Pull Request</h1>

<h2 style="font-weight:bold" align="center">✅Development Checklist✅</h2>

<details open><summary>🌳Branching🌳</summary>

Does the name of the head(source) branch for this pull request have the correct **_preview feature branch_** syntax?

Syntax: _preview/feature/&lt;issue-id&gt;-&lt;description&gt;_  
Example: _preview/feature/123-my-preview-change_
- [x] Yes
- [ ] No
 
Does the name of the base(target) branch for this pull request have the correct **_preview branch_** syntax?

Syntax: _preview/v&lt;major-number&gt;.&lt;minor-number&gt;.&lt;patch-number&gt;-preview.&lt;prev-number&gt;_  
Example: _preview/v1.2.3-preview.4_
- [x] Yes
- [ ] No

Is the head(source) branch for this pull request created from a branch with the correct **_preview branch_** syntax?

Syntax: _preview/v&lt;major-number&gt;.&lt;minor-number&gt;.&lt;patch-number&gt;-preview.&lt;prev-number&gt;_  
Example: _preview/v1.2.3-preview.4_
- [x] Yes
- [ ] No
</details>

<details open><summary>🐛Bugs🐛</summary>

Contains Bug Fix(es)
- [ ] Yes
    - [ ] A bug-label label has been added to the pull request.
- [x] No
</details>

<details open><summary>💣Breaking Change(s)💣</summary>

Contains [breaking change(s)](https://docs.microsoft.com/en-us/dotnet/core/compatibility/#modifications-to-the-public-contract)
- [x] Yes
    - [x] A breaking-change-label label has been added to the pull request.
- [ ] No
</details>

<details open><summary>✨Enhancements✨</summary>

Contains enhancement(s) that add a feature or behavior.
- [ ] Yes
    - [ ] An enhancement-label label has been added to the pull request.
- [x]  No
</details>

<details open><summary>⚙️Workflow (CI/CD) Changes⚙️</summary>

Contains changes to workflow files. These changes can only done by the project maintainers.
- [ ] Yes
    - [ ] A workflow-label label has been added to the pull request.
- [x] No
</details>

<details open><summary>📃Documentation Updates📃</summary>

Contains changes that require documentation updates to code docs or **Velaptor** documentation.
- [ ] Yes
    - [ ] I have updated the documentation accordingly.
    - [ ] A documentation-label label has been added to the pull request.
- [x] No
</details>

<details open><summary>🧪Unit Testing🧪</summary>

My change requires unit tests to be written.
- [x] Yes
- [ ] No
</details>

<details open><summary>🧪Manual Testing🧪</summary>

I have manually tested my changes. (This can be done by using project named **_VelaptorTesting_**).
- [x] Yes
- [ ] No
</details>

<h2 style="font-weight:bold" align="center">✅Code Review Checklist✅</h2>

- [x] Pull request title matches the title of the linked issue.
- [x] Associated issue exists and is linked to this pull request.
    - One issue per pull request.
- [x] My code follows the coding style of this project.
    - This is enforced by the *.editorconfig* files in the project and displayed as warnings.  If there is an edge case with coding style that should be ignored or changed, reach out and let's discuss it.
- [x] All tests passed locally.
    - Status checks are put in place to run unit tests every single time a change is pushed to a pull request.  This does not mean that the tests pass in both the local and CI environment.
- [x] A preview-label label has been added to the pull request.

### CODE REVIEW

**Comment by CalvinWilkinson** [comment]
This is a breaking change due to the change of the exception message for null or empty strings `ArgumentNullException`s that are thrown throughout the code base.

File: Testing/VelaptorTests/Guards/EnsureThatTests.cs [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    using Xunit;
+
+    /// <summary>
+    /// Tests the <see cref="EnsureThat"/> class.
+    /// </summary>
+    public class EnsureThatTests
+    {
+        #region Method Tests
+        [Fact]
+        public void ArgIsNotNull_WithNullValue_ThrowsException()
```
Comment (CalvinWilkinson) [inline Testing/VelaptorTests/Guards/EnsureThatTests.cs:19]: Rename the first section of this unit test method to match the method under test

File: Testing/VelaptorTests/Guards/EnsureThatTests.cs [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+            // Act & Assert
+            AssertExtensions.ThrowsWithMessage<ArgumentNullException>(() =>
+            {
+                EnsureThat.ParamIsNotNull(nullObj);
+            }, "The parameter must not be null. (Parameter 'nullObj')");
+        }
+
+        [Fact]
+        public void ThrowIfArgIsNull_WithNonNullValue_DoesNotThrowException()
```
Comment (CalvinWilkinson) [inline Testing/VelaptorTests/Guards/EnsureThatTests.cs:32]: Rename the first section of this unit test method to match the method under test

File: Testing/VelaptorTests/Guards/EnsureThatTests.cs [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+            AssertExtensions.DoesNotThrow<Exception>(() =>
+            {
+                EnsureThat.ParamIsNotNull(nonNullObj);
+            });
+        }
+
+        [Theory]
+        [InlineData("")]
+        [InlineData(null)]
+        public void StringArgIsNotNullOrEmpty_WhenInvoked_ThrowsException(string value)
```
Comment (CalvinWilkinson) [inline Testing/VelaptorTests/Guards/EnsureThatTests.cs:47]: Rename the first section of this unit test method to match the method under test

**Review by CalvinWilkinson — COMMENTED** [review]
Changes required
